packages/jest-circus/src/utils.ts

```diff
@@ -212,6 +216,15 @@ export const callAsyncCircusFn = (
 
           return reason ? reject(errorAsErrorObject) : resolve();
         });
+
+        if (testOrHook.seenDone) {
```

Comment:
why do it down here?

Reply:
We want to first check if `done` was passed with a `reason` for failure, if so, return the `reason`. Or else, we can check if `done` was called more than once and return that as an error. We have some tests that checked if `reason` was being sent out properly, and they were failing because we returned our ` Expected done to be called once, but it was called multiple times.` error before the reason was checked.

Reply:
Which tests are those? If they call `done` multiple times that's wrong. I can take a look if there are cases it does make sense though - could you point me to the tests that failed?

Reply:
Here's one such test that calls `done` more than once.

https://github.com/facebook/jest/blob/591d612f6ff0a5f8769cbe7618d3d8f507bcbb81/packages/jest-core/src/__tests__/SearchSource.test.ts#L481-L498

Reply:
Ok, I was confused about something else, please ignore this. I will revert the position-change commit. However, this doesn't work as expected: https://github.com/facebook/jest/blob/3641e3cc7309caf4c755add40bebcb77fc18b786/e2e/failures/__tests__/errorAfterTestComplete.test.js#L10-L13
Expects: `Error: Caught error after test environment was torn down`
Receives: `Expected done to be called once, but it was called multiple times.`